==================================== INPUT ====================================
# expected-to-throw
query sameFieldsWithDifferentArgs_MeQuery {
  me {
    name
    address {
      city
    }
    ...sameFieldsWithDifferentArgs_NestedFragment
    ...sameFieldsWithDifferentArgs_NestedFragment2
  }
}

fragment sameFieldsWithDifferentArgs_NestedFragment on User {
  firstName(if: false, unless: true)
}

fragment sameFieldsWithDifferentArgs_NestedFragment2 on User {
  firstName(if: false, unless: false)
}
==================================== ERROR ====================================
âœ–ï¸Ž Expected all fields on the same parent with the name or alias `firstName` to have the same argument values after applying fragment arguments. This field has the applied argument values: (if: false, unless: true)

  same-fields-with-different-args.invalid.graphql:14:3
   13 â”‚ fragment sameFieldsWithDifferentArgs_NestedFragment on User {
[1m   14 â”‚ [0m  [31mf[0m[31mi[0m[31mr[0m[31ms[0mtName(if: false, unless: true)
      â”‚ [31m  ^^^^^^^^^[0m
[1m   15 â”‚ [0m}

[31m  â„¹ï¸Ž which conflicts with this field with applied argument values (if: false, unless: false)[0m

  [4msame-fields-with-different-args.invalid.graphql[0m[2m:18:3[0m
[1m   17 â”‚ [0mfragment sameFieldsWithDifferentArgs_NestedFragment2 on User {
   18 â”‚   firstName(if: false, unless: false)
      â”‚ [31m  ^^^^^^^^^[0m
[1m   19 â”‚ [0m}
